### PR TITLE
Fix the API for adding decorated `ServiceWithPathMappings` and update…

### PR DESCRIPTION
**Pull request by trustin**
… documentation

Motivation:

We recently received a good question about how to decorate a
`GrpcService`, which is a `ServiceWithPathMappings`.

I found there is not enough explanation about such a question and it is
currently not easy to specify multiple decorators for a
`ServiceWithPathMappings`.

Modifications:

- Allow specifying multiple decorators when registering a
  `ServiceWithPathMappings`.
- Add some sections about decorating `ServiceWithPathMappings` and
  `GrpcService`.

Result:

- Easier to decorate a `ServiceWithPathMappings` such as `GrpcService`.
- Better documentation.

### CODE REVIEW

**Comment by trustin** [comment]
@perlun Looking forward to your feed back on the documentation.

File: site/src/sphinx/server-decorator.rst [hunk 1]
```diff
@@ -135,6 +135,45 @@ a certain type from a server:
...
+.. _server-decorator-service-with-path-mappings:
+
+Decorating ``ServiceWithPathMappings``
+--------------------------------------
+
+:api:`ServiceWithPathMappings` is a special variant of :api:`Service` which allows a user to register multiple
+routes for a single service easily. Because it requires a special treatment unlike usual :apiplural:`Service`,
+the two following restrictions apply:
+
+- You cannot specify path mappings when registering a :api:`ServiceWithPathMappings`.
```
Comment (anuraaga) [inline site/src/sphinx/server-decorator.rst]: Might want to clarify that actually they can, but it won't use the `ServiceWithPathMappings` in such a case.

File: site/src/sphinx/server-grpc.rst [hunk 2]
```diff
@@ -93,6 +93,30 @@ a :api:`GrpcServiceBuilder` and add it to the :api:`ServerBuilder`:
     We bound the :api:`GrpcService` without specifying any path mappings. It is because :api:`GrpcService`
     implements :api:`ServiceWithPathMappings`, which dynamically provides path mappings by itself.
 
+.. _server-grpc-decorator:
+
+Decorating a ``GrpcService``
+----------------------------
+
+Unlike a usual Armeria :api:`Service`, :api:`GrpcService` implements a special interface called
+:api:`ServiceWithPathMappings`. You cannot decorate a :api:`GrpcService` using ``Service.decorate()`` but
```
Comment (anuraaga) [inline site/src/sphinx/server-grpc.rst]: Ditto, `cannot` is a bit confusing since it is actually possible, but only when using an explicit path mapping.

File: site/src/sphinx/server-decorator.rst (hunk 1 above)
Comment (trustin) [reply site/src/sphinx/server-decorator.rst]: Ah, yeah. Let me rephrase it a little bit.

Comment (trustin) [reply site/src/sphinx/server-decorator.rst]: Updated. PTAL again :-)

File: site/src/sphinx/server-decorator.rst [hunk 3]
```diff
@@ -135,6 +135,70 @@ a certain type from a server:
...
+Decorating ``ServiceWithPathMappings``
+--------------------------------------
+
+:api:`ServiceWithPathMappings` is a special variant of :api:`Service` which allows a user to register multiple
+routes for a single service. It has a method called ``pathMappings()`` which returns a ``Set`` of
+:apiplural:`PathMapping` so that you do not have to specify path mappings when registering your service:
+
+.. code-block:: java
+
+    public class MyServiceWithPathMappings implements ServiceWithPathMappings<HttpRequest, HttpResponse> {
```
Comment (minwoox) [inline site/src/sphinx/server-decorator.rst]: Seems like we missed some import statements throughout this document.
Such as for `ServiceWithPathMappings`, `ServiceRequestContext` and `PathMapping`.

Comment (trustin) [reply site/src/sphinx/server-decorator.rst]: Thanks for the heads up. Added.

**Comment by minwoox** [comment]
LGTM one more. 😄

**Comment by trustin** [comment]
Thanks for reviewing.
